### PR TITLE
instructor reject action for email issue

### PR DESCRIPTION
**Pull request by ibrahim-kardi**
instructor rejection new action added, and email not  sending instructor rejection issue fixed

### CODE REVIEW

File: views/pages/instructors.php [hunk 1]
```diff
@@ -46,9 +46,19 @@
...
 // Available status for instructor.
+$available_status = array(
+	'pending'  => array( __( 'Pending', 'tutor' ), 'select-warning' ),
+	'approved' => array( __( 'Approved', 'tutor' ), 'select-success' ),
+	'blocked'  => array( __( 'Blocked', 'tutor' ), 'select-danger' ),
+);
+
 $instructor_status = array( 'approved', 'pending', 'blocked' );
 if ( 'pending' === $active_tab ) {
 	$instructor_status = array( 'pending' );
```
Comment (harunollyo) [inline views/pages/instructors.php:57]: Simply add it like as below
```php
$instructor-action['reject'] = array( __( 'Reject', 'tutor' ), 'select-danger' );
```

File: classes/Instructors_List.php [hunk 2]
```diff
@@ -207,6 +207,14 @@ public static function update_instructors( $status, $user_ids ): bool {
 				self::remove_instructor_role( $instructor_id, $status );
 			}
 		}
+		if ( 'reject' === $status ) {
+			$arr = explode( ',', $user_ids );
+			foreach ( $arr as $instructor_id ) {
+				$instructor_id = (int) sanitize_text_field( $instructor_id );
+				self::remove_instructor_role( $instructor_id, $status );
+			}
+			do_action( 'tutor_after_rejected_instructor', $instructor_id );
```
Comment (harunollyo) [inline classes/Instructors_List.php:216]: action trigger should not be outside of the loop.

File: classes/Instructors_List.php [hunk 3]
```diff
@@ -207,6 +207,14 @@ public static function update_instructors( $status, $user_ids ): bool {
 				self::remove_instructor_role( $instructor_id, $status );
 			}
 		}
+		if ( 'reject' === $status ) {
+			$arr = explode( ',', $user_ids );
+			foreach ( $arr as $instructor_id ) {
+				$instructor_id = (int) sanitize_text_field( $instructor_id );
+				self::remove_instructor_role( $instructor_id, $status );
```
Comment (harunollyo) [inline classes/Instructors_List.php:214]: Remove the hooks from this method and make action specific method by wrapping this method.

**Review by harunollyo — CHANGES_REQUESTED** [review]

File: views/pages/instructors.php (hunk 1 above)
Comment (ibrahim-kardi) [reply views/pages/instructors.php]: updated @harunollyo vai

File: classes/Instructors_List.php [hunk 4]
```diff
@@ -204,7 +204,14 @@ public static function update_instructors( $status, $user_ids ): bool {
 			$arr = explode( ',', $user_ids );
 			foreach ( $arr as $instructor_id ) {
 				$instructor_id = (int) sanitize_text_field( $instructor_id );
-				self::remove_instructor_role( $instructor_id, $status );
+				self::instructor_blockage( $instructor_id, $status );
```
Comment (harunollyo) [inline classes/Instructors_List.php:207]: Remove the status param and pass the param statically from `instructor_blockage` method.

**Review by harunollyo — CHANGES_REQUESTED** [review]

**Review by harunollyo — APPROVED** [review]

**Pull request merged**